### DISCUSSION TITLE
Discord OAuth2

### DISCUSSION THREAD

**Discussion by Subie1** [pocketbase/pocketbase#2498] [Q&A] [answered]
I've been using the Discord OAuth2 but I noticed that you can't access the Discord API with the token pocketbase provided the data you can access consists of Email (if the user has one), username & discord id.

I'm trying to access more data using the Discord API for example the Guilds the user has joined and no API endpoinds suffice with the token provided.

I use PocketBase: 0.15.3
and ReactJS

Authentication Code:

```js
await pb.collection("users").authWithOAuth2({ provider: "discord" });
```

Code to access API:
```js
axios.get("https://discord.com/api/users/@me/guilds", { headers: { authentication: `Bearer ${pb.authStore.token}` }});
```

The code returns a 401 response indicating the token is invalid.

**Comment by ganigeorgiev** [comment, marked as answer]
`pb.authStore.token` is the PocketBase authentication token. The Discord OAuth2 access and refresh tokens can be accessed after OAuth2 authentication under the `meta` prop, eg:

```js
const authData = await pb.collection("users").authWithOAuth2({ provider: "discord" });

console.log(authData.meta.accessToken);
```

**Reply by ganigeorgiev** [reply]
Also note that you can specify your own custom scopes via the `scopes` config option:

```js
await pb.collection("users").authWithOAuth2({
    provider: "discord",
    scopes:   ["identify", "email", "yourCustomScope..."]
});
```
